torchvision/prototype/datasets/_builtin/mnist.py

```diff
@@ -0,0 +1,396 @@
+            valid_options=dict(
+                split=("train", "test", "test10k", "test50k", "nist"),
+            ),
+        )
+
+    _URL_BASE = "https://raw.githubusercontent.com/facebookresearch/qmnist/master"
+    _CHECKSUMS = {
+        "qmnist-train-images-idx3-ubyte.gz": "9e26a7bf1683614e065d7b76460ccd52807165b3f22561fb782bd9f38c52b51d",
+        "qmnist-train-labels-idx2-int.gz": "2c05dc77f6b916b38e455e97ab129a42a444f3dbef09b278a366f82904e0dd9f",
+        "qmnist-test-images-idx3-ubyte.gz": "43fc22bf7498b8fc98de98369d72f752d0deabc280a43a7bcc364ab19e57b375",
+        "qmnist-test-labels-idx2-int.gz": "9fbcbe594c3766fdf4f0b15c5165dc0d1e57ac604e01422608bb72c906030d06",
+        "xnist-images-idx3-ubyte.xz": "f075553993026d4359ded42208eff77a1941d3963c1eff49d6015814f15f0984",
+        "xnist-labels-idx2-int.xz": "db042968723ec2b7aed5f1beac25d2b6e983b9286d4f4bf725f1086e5ae55c4f",
+    }
+
+    def _files_and_checksums(self, config: DatasetConfig) -> Tuple[Tuple[str, str], Tuple[str, str]]:
+        prefix = "xnist" if config.split == "nist" else f"qmnist-{'train' if config.split== 'train' else 'test'}"
+        suffix = "xz" if config.split == "nist" else "gz"
+        images_file = f"{prefix}-images-idx3-ubyte.{suffix}"
+        labels_file = f"{prefix}-labels-idx2-int.{suffix}"
+        return (images_file, self._CHECKSUMS[images_file]), (
+            labels_file,
+            self._CHECKSUMS[labels_file],
+        )
+
+    def _collate_and_decode(
+        self,
+        data: Tuple[np.ndarray, np.ndarray],
+        *,
+        config: DatasetConfig,
+        decoder: Optional[Callable[[io.IOBase], torch.Tensor]],
+    ):
+        image_array, label_array = data
+        label_parts = label_array.tolist()
+        sample = super()._collate_and_decode((image_array, label_parts[0]), config=config, decoder=decoder)
+
+        sample.update(
+            dict(
+                zip(
+                    ("nist_hsf_series", "nist_writer_id", "digit_index", "nist_label", "global_digit_index"),
+                    label_parts[1:6],
+                )
+            )
+        )
+        sample.update(dict(zip(("duplicate", "unused"), [bool(value) for value in label_parts[-2:]])))
+        return sample
+
+    def _make_datapipe(
+        self,
+        resource_dps: List[IterDataPipe],
+        *,
+        config: DatasetConfig,
+        decoder: Optional[Callable[[io.IOBase], torch.Tensor]],
+    ) -> IterDataPipe[Dict[str, Any]]:
+        dp = super()._make_datapipe(resource_dps, config=config, decoder=decoder)
+        if config.split not in ("test10k", "test50k"):
+            return dp
+
+        start: Optional[int]
+        stop: Optional[int]
+        if config.split == "test10k":
+            start = 0
+            stop = 10000
+        else:  # config.split == "test50k"
+            start = 10000
+            stop = None
+
+        return Slicer(dp, start=start, stop=stop)
```

Comment:
Is there way we can untackle `test10k` and `test50k` at the resource stage?

Reply:
Unfortunately, no. The earliest would be in the `MNISTFileReader` that could enumerate the chunks and discard the unused chunks directly. In fact, that would improve performance since otherwise we also decode the chunks that we throw away later. I'll send a patch.

Reply:
The latest commits include this change